apps/assets/migrations/0016_directory_service.py

```diff
@@ -0,0 +1,59 @@
+# Generated by Django 4.1.13 on 2025-04-03 09:51
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("assets", "0015_automationexecution_type"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="DirectoryService",
+            fields=[
+                (
+                    "asset_ptr",
+                    models.OneToOneField(
+                        auto_created=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        parent_link=True,
+                        primary_key=True,
+                        serialize=False,
+                        to="assets.asset",
+                    ),
+                ),
+                (
+                    "domain_name",
+                    models.CharField(
+                        blank=True,
+                        default="",
+                        max_length=128,
+                        verbose_name="Domain name",
+                    ),
+                ),
+            ],
+            options={
+                "verbose_name": "Directory service",
+            },
+            bases=("assets.asset",),
+        ),
+        migrations.AddField(
+            model_name="platform",
+            name="ds_enabled",
+            field=models.BooleanField(default=False, verbose_name="DS enabled"),
+        ),
+        migrations.AddField(
+            model_name="platform",
+            name="ds",
+            field=models.ForeignKey(
+                blank=True,
+                null=True,
+                on_delete=django.db.models.deletion.SET_NULL,
+                related_name="ds_platforms",
+                to="assets.directoryservice",
+                verbose_name="Active Directory",
+            ),
+        ),
+    ]
```

Comment:
The code does not seem to contain any issues. No changes or modifications have been made since the last known update.

One suggestion could be to ensure that there is an error management mechanism implemented if errors occur during execution of this migration process. Also, consider implementing unit test cases for this migration in case of unexpected behaviors when it's deployed to a production environment.